lib/Adapter/WorseReflection/Transformer/AddMissingProperties.php

```diff
@@ -75,7 +74,7 @@ public function transform(SourceCode $code): SourceCode
         return SourceCode::fromString($code);
     }
 
-    private function resolveClassBuilder(SourceCodeBuilder $sourceBuilder, ReflectionClassLike $class): ClassLikeBuilder
+    private function resolveClassBuilder(SourceCodeBuilder $sourceBuilder, ReflectionClassLike $class)
```

Comment:
`trait` and `class` do indeed both extend `ClassLikeBuilder`, and both the `TraitBuilder` and `ClassBuilder` classes implement the `property` method too. It's only `InterfaceBuilder` that doesn't

Reply:
We could keep the return type and instead inform `phpstan` that it is one of `TraitBuilder` or `ClassBuilder`:

```
assert($builder instanceof TraitBuilder || $builder instanceof ClassBuilder);
```
after we call the method. (another alternative would  be `/** @var TraitBuilder|ClassBuilder $builder */` but `assert` is nicer imo.

This let's phpstan analyse the code better, so it's arguably a good thing.

Reply:
Ooo, I like that. Tested with adding `@return TraitBuilder | ClassBuilder` in docblock, and it makes Stan very happy indeed.

If we were to go the assert route, where would it go in the code? I've never seen it used outside of tests and can't find anything about it in phpstan's readme.

Reply:
Ok, I've just pushed the method docblock. I didn't notice the existing `/** @var ReflectionClass $class */` above the foreach around line 35 before this, however with method docblock we also have the advantage of only ever having to hint it once. Or would phpstan "remember" the first `@var` hint if it was later reused in the file?

Reply:
That's fine, basically `assert` can be used to provide type information within the AST, so we don't rely on docblocks.

It would look like:

```php
$builder = $this->resolveBuilder($className);
assert($builder instanceof Foobar || $builder instanceof Barfoo)
```

But the docblock is fine here :+1:

Reply:
Ahh makes perfect sense now! Thanks for the clarification :slightly_smiling_face: